framework/src/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -949,146 +951,175 @@ public Void visitMethodInvocation(MethodInvocationTree node, Void p) {
     }
 
     /**
-     * Checks all the preconditions of the method invocation or variable access {@code tree} with
-     * element {@code invokedElement}.
+     * Checks all the preconditions of the method invocation {@code tree} with
+     * element {@code invokedMethodElement}.
+     *
+     * @param methodInvocationTree the MethodInvocationTree to check preconditions for.
+     * @param invokedMethodElement the ExecutableElement to retrieve the method preconditions from.
      */
-    protected void checkPreconditions(Tree tree,
-            Element invokedElement, boolean methodCall) {
-        checkPreconditions(tree, invokedElement, methodCall, null);
+    protected void checkMethodInvocationPreconditions(MethodInvocationTree methodInvocationTree,
+            ExecutableElement invokedMethodElement) {
+        if (invokedMethodElement == null) {
+            return;
+        }
+
+        checkPreconditions(methodInvocationTree, contractsUtils.getPreconditions(invokedMethodElement));
     }
 
     /**
-     * Checks all the preconditions of the method invocation or variable access {@code tree} with
-     * element {@code invokedElement}.
+     * Checks that all the given {@code preconditions} hold true immediately prior to
+     * the method invocation or variable access at {@code tree}.
```

Comment:
If you would prefer the term 'precondition' not to be used because it makes users think of `@PreconditionAnnotation`, we could use a different term.